packages/web3-utils/src/converters.ts

```diff
@@ -411,6 +411,13 @@ export const toHex = (
  */
 export const toNumber = (value: Numbers): number | bigint => {
 	if (typeof value === 'number') {
+            if (value > 100000000000000000000) {
```

Comment:
Tried to use this code but receive incorrect result
```toNumber(100000000000000000010)
returns  100000000000000000000
```

Reply:
This is a behaviour built in javascript number system. This can not be fixed.